Update `list`, `find` and `sort` commands

JeremyYong128 opened the pull request:
* `list` now removes and filter but preserves sort order.
* `find` now supports searching by multiple fields. All keywords provided must match.
* `find` and `sort` commands can be done on all fields.

Tests are not implemented yet.

JeremyYong128 committed:
Update `list`, `find` and `sort` commands

* `list` now removes and filter but preserves sort order.
* `find` now supports searching by multiple fields. All keywords
provided must match.
* `find` and `sort` commands can be done on all fields.

Tests are not implemented yet.

evanyan13 commented on src/main/java/seedu/application/logic/Messages.java:
```diff
@@ -14,10 +14,11 @@
     public static final String MESSAGE_INVALID_JOB_DISPLAYED_INDEX = "The job index provided is invalid!";
-    public static final String MESSAGE_INVALID_SPECIFIER = "The field specifier provided is invalid!";
-    public static final String MESSAGE_JOBS_LISTED_OVERVIEW = "%1$d jobs listed!";
+    public static final String MESSAGE_INVALID_PREFIX = "The prefix provided is invalid!";
```
Can I clarify the difference between the meaning btw "Prefix" and "Field Specifier"? It is correct if I understand as this:
- Prefix is targeted for the field -> c/ r/ etc
- Field Specifier is for the command -> -c -r etc

so technically if I specifier -c-r in the command input, I must have c/ r/ in the subsequent entry of the command?

evanyan13 commented on src/main/java/seedu/application/logic/parser/FindCommandParser.java:
```diff
@@ -21,34 +29,49 @@ public class FindCommandParser implements Parser<FindCommand> {
+
+        ArrayList<FieldContainsKeywordsPredicate> predicateList = new ArrayList<>();
+
+        for (Map.Entry<Prefix, List<String>> entry : argMultimap.getArgMultimap().entrySet()) {
```
Good implementation of Find function using a hashmap 👍

evanyan13 commented on src/main/java/seedu/application/logic/parser/FindCommandParser.java:
```diff
@@ -21,34 +29,49 @@ public class FindCommandParser implements Parser<FindCommand> {
         }
 
-        return keywords;
+        return new FindCommand(new CombinedPredicate(predicateList));
```
Good use of another class `CombinedPredicate` to handle the operation with the `predicateList`

evanyan13 commented on src/main/java/seedu/application/model/job/Industry.java:
```diff
@@ -9,6 +9,7 @@
  */
 
 public class Industry {
+    public static final String INDUSTRY_SPECIFIER = "-i";
```
Hmm do you think it will be better to place all the Specifiers in the `CliSyntax` class? Or in another class by itself?

In similar sense, do you think it will be a good idea to implement another class like `Specifier.java` to handle the operations w field specifiers just like `Prefix.java`

evanyan13 reviewed (approved):
LGTM! Thank you for the enhancement and refinement of the current list, find and sort commands. Left some comments for thoughts 👍

JeremyYong128 committed:
Remove specifiers

Prefixes are now used instead of specifiers for the `sort` command.
This reduces confusion in users regarding the differences between a specifier and a prefix.

evanyan13 reviewed (approved):
LGTM
